File: exam-c-lang/01_word_count.py
import sys

IN = 1
OUT = 0

state = OUT
line_count = word_count = letter_count = 0

for line in sys.stdin:
    for c in line:
        if c == ' ' or c == '\t' or c == '\n':
            state = OUT
        elif not state:
            state = IN
            word_count += 1
        letter_count += 1
    line_count += 1

print(line_count, word_count, letter_count)

File: exam-c-lang/01_word_count.c
#include <stdio.h>

#define IN 1
#define OUT 0

int main()
{
    int state, c, line_count, word_count, letter_count;

    state = OUT;
    line_count = word_count = letter_count = 0;
    while ((c = getchar()) != EOF)
    {
        letter_count++;
        if (c == '\n')
            line_count++;
        if (c == ' ' || c == '\t' || c == '\n')
            state = OUT;
        else if (state == OUT)
        {
            state = IN;
            word_count++;
        }
    }

    printf("%d %d %d\n", line_count, word_count, letter_count);

    return 0;
}